src/services/session.service.ts

```diff
@@ -62,6 +62,17 @@ export class SessionService {
     return token;
   }
 
+  async authenticateFacebook(userData: User) {
+    try {
+      const user = await this.userService.findOrCreateUserFacebook(userData);
+      const token = await this.userService.generateToken(user);
+      return token;
+    } catch (error) {
+      throw new Error(`${error}`);
```

Comment:
What happens if your user service throws a custom exception? Are you sure you are not nesting exceptions here?